### PR TITLE
FEATURE: ping for testing webhooks

### PR DESCRIPTION
**Pull request by fantasticfears**

### CODE REVIEW

File: app/controllers/admin/web_hooks_controller.rb [hunk 1]
```diff
@@ -92,6 +92,38 @@ def redeliver_event
...
+      'Host' => uri.host,
+      'User-Agent' => "Discourse/" + Discourse::VERSION::STRING,
+      'X-Discourse-Event-Id' => 'null',
+      'X-Discourse-Event-Type' => 'ping'
+    }
+
+    if @web_hook.secret.present?
+      headers['X-Discourse-Event-Signature'] = "sha256=" + OpenSSL::HMAC.hexdigest("sha256", @web_hook.secret, body)
+    end
+    conn.post(headers: headers, body: body)
```
Comment (tgxworld) [inline app/controllers/admin/web_hooks_controller.rb]: Hmm it doesn't seem like we are checking the response to see if the ping fails?

File: app/assets/javascripts/admin/controllers/admin-web-hooks-show-events.js.es6 [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+import { popupAjaxError } from 'discourse/lib/ajax-error';
+
 export default Ember.Controller.extend({
   actions: {
     loadMore() {
       this.get('model').loadMore();
+    },
+
+    ping() {
+      return bootbox.confirm(I18n.t('admin.web_hooks.events.ping_instruction'), I18n.t('no_value'), I18n.t('yes_value'), result => {
```
Comment (tgxworld) [inline app/assets/javascripts/admin/controllers/admin-web-hooks-show-events.js.es6]: A ping event is pretty safe so I don't think we need to ask for confirmation.

**Comment by tgxworld** [comment]
Aside from some minor comments. This PR looks really good to me

File: app/controllers/admin/web_hooks_controller.rb (hunk 1 above)
Comment (fantasticfears) [inline app/controllers/admin/web_hooks_controller.rb]: Nope, it should be used to test whether the configuration/network is functional.

Comment (tgxworld) [inline app/controllers/admin/web_hooks_controller.rb]: If it isn't functional.. How will they know whether it is their configuration or a bug in the code?

Comment (fantasticfears) [inline app/controllers/admin/web_hooks_controller.rb]: Fixed 👍

**Pull request merged**